src/global_distrib/mod_global_distrib_mapping.erl

```diff
@@ -298,3 +302,14 @@ do_lookup_jid({_, _, _} = Jid) ->
                 BareJid -> ets_cache:lookup(?JID_TAB, BinJid, LookupInDB(jid:to_binary(BareJid)))
             end
     end.
+
+-spec get_advertised_endpoints(Opts :: [{Key :: term(), Value :: term()}]) -> [endpoint()].
+get_advertised_endpoints(Opts) ->
```

Comment:
get_endpoints/1 returns `{ok, [endpoints()]}`.
But this get_adverised_endpoints/1 returns `[endpoint()]` or false (not in the spec, expect warning from dialyzer).

I would like to be more consistent, even if the first argument has different type.

 I am feeling like adding gen_mod:opts() type is a good idea, instead of `[{Key :: term(), Value :: term()}]`.

Reply:
So, for now, I will make `Opts` a `list()` as it's done in similar situations (ie. in [mod_mam](https://github.com/esl/MongooseIM/blob/master/src/mod_mam.erl#L178)). We can think about introducing such a type but as it requires changing this type in the whole src/ I'll leave this job for now. How about that?

Reply:
I also weren't thinking about consistency of the result of this function cause it's not even exported. I'll change it though :)

Reply:
Are you ok with just changing Opts type to `list()` for now?